instrumentation/opentelemetry-annotations-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/otelannotations/WithSpanTracer.java

```diff
@@ -23,19 +25,33 @@ public static WithSpanTracer tracer() {
 
   private static final Logger log = LoggerFactory.getLogger(WithSpanTracer.class);
 
+  private final MethodSpanStrategies methodSpanStrategies = MethodSpanStrategies.getInstance();
+
   public Context startSpan(
       Context parentContext, WithSpan applicationAnnotation, Method method, SpanKind kind) {
+
+    Context spanStrategyContext = withMethodSpanStrategy(parentContext, method);
```

Comment:
Hmm, do you really need to store the strategy in the context? I mean, you could just call it directly in the `WithSpanAdvice#stopSpan()` method using the returned value.

Reply:
Agreed, I've removed the use of context here.